Controller/CredentialsChecker/Index.php

```diff
@@ -90,30 +88,11 @@ public function execute()
     {
         if (($params = $this->getRequest()->getParams())
             && (!empty($params['secretKey']) && !empty($params['merchantKey']))) {
-            if (preg_match('/[^\*]/', $params['secretKey'])) {
-                $secretKey = $params['secretKey'];
-            } else {
-                $secretKey = $this->encryptor->decrypt($this->configProviderAccount->getSecretKey());
-            }
-
-            if (preg_match('/[^\*]/', $params['merchantKey'])) {
-                $merchantKey = $params['merchantKey'];
-            } else {
-                $merchantKey = $this->encryptor->decrypt($this->configProviderAccount->getMerchantKey());
-            }
-
-            $mode = $params['mode'] ?? Data::MODE_TEST;
-
-            if (!$this->testJson($mode, $merchantKey, $secretKey, $message)) {
+            if($this->client->confirmCredential()){
```

Comment:
It is not correct because the secretKey and merchantKey is not set yet on Configuration 

add as parameter secretKey and merchantKey and if they are null used the config setting